src/wallet/wallet_rpc_server.cpp

```diff
@@ -615,22 +622,47 @@ namespace tools
         return false;
       }
 
-      if (!req.do_not_relay)
-        m_wallet->commit_tx(ptx_vector);
-
-      // populate response with tx hash
-      res.tx_hash = epee::string_tools::pod_to_hex(cryptonote::get_transaction_hash(ptx_vector.back().tx));
-      if (req.get_tx_key)
+      if (m_wallet->multisig())
       {
-        res.tx_key = epee::string_tools::pod_to_hex(ptx_vector.back().tx_key);
+        res.multisig_txset = epee::string_tools::buff_to_hex_nodelimer(m_wallet->save_multisig_tx(ptx_vector));
+        if (res.multisig_txset.empty())
+        {
+          er.code = WALLET_RPC_ERROR_CODE_UNKNOWN_ERROR;
+          er.message = "Failed to save multisig tx set after creation";
+          return false;
+        }
       }
-      res.fee = ptx_vector.back().fee;
-
-      if (req.get_tx_hex)
+      else
       {
-        cryptonote::blobdata blob;
-        tx_to_blob(ptx_vector.back().tx, blob);
-        res.tx_blob = epee::string_tools::buff_to_hex_nodelimer(blob);
+        if (!req.do_not_relay)
+          m_wallet->commit_tx(ptx_vector);
+
+        // populate response with tx hash
+        res.tx_hash = epee::string_tools::pod_to_hex(cryptonote::get_transaction_hash(ptx_vector.back().tx));
+        if (req.get_tx_key)
```

Comment:
Shouldn't this and others `req.get_tx_hex ` and `req.get_tx_metadata ` be also done with the multisig case?

Reply:
Doesn't make sense in this case, as the tx is partly made and the metadata is something used to send a previously made tx.

Reply:
I agree with you regarding the tx hex and I'm not familiar enough with the usage of the tx metadata to have an opinion. Still, the tx secret key is already set at this point, and it should be reasonable to return it when `req.get_tx_key` is set to `true`. Otherwise, I don't see any RPC command that returns a tx secret key for multisig.

Reply:
Yes, tx key makes sense.